base/docs/Docs.jl

```diff
@@ -239,11 +239,27 @@ end
 # =================
 
 """
+    getdoc(x::T, sig) -> String
+
+Return the dynamic docstring associated with object `x`, or `nothing` to use
+the binding's documentation.
+"""
+getdoc(x, sig) = getdoc(x)
+getdoc(x) = nothing
+
+"""
     Docs.doc(binding, sig)
 
 Returns all documentation that matches both `binding` and `sig`.
+
+If `getdoc` returns a non-`nothing` result on the value of the binding, then a
+dynamic docstring is returned instead of one based on the binding itself.
 """
 function doc(binding::Binding, sig::Type = Union{})
+    if defined(binding)
+        result = getdoc(resolve(binding), sig)
+        result === nothing || return Markdown.parse(result)
```

Comment:
Would it make sense to *not* call `Markdown.parse` here and instead just return `result`? That way `getdocs` could return a more suitable object, such as an `RST` object for rst-formatted python docstrings.